### DISCUSSION TITLE
Is there compile option for generating message struct containing enumeration use enum-type version by default, instead of `i32`?

### DISCUSSION THREAD

**Discussion by alissa-tung** [tokio-rs/prost#1146] [Q&A] [answered]
destruct and construct those messages with i32 filed are annoying, and unsafe in someway

**Comment by caspermeijn** [comment, marked as answer]
The short answer is: no. 

`prost-build` will generate a getter for each enum field, which returns the enum type.

There are multiple issues and PR related to this. If you are interested in why this is the status quo, I suggest you read those. 

Do you want to work on improving the generated API? If so, please contact me.

**Reply by alissa-tung** [reply]
thank you!
